actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -134,7 +134,7 @@ module CacheHelper
       #
       #   <%= render @notifications, cache: false %>
       def cache(name = {}, options = nil, &block)
-        if controller.perform_caching
+        if controller.try(:perform_caching)
```

Comment:
#try can also be used for it's secondary effect of "receiver is there but may or may not respond to the message". Seems like legit usage here.